### DISCUSSION TITLE
Path to Picture

### DISCUSSION THREAD

**Discussion by BerndDemian** [typst/typst#2436] [General]
I cannot solve the following problem.
I have a few convenience functions, including for displaying images. The pictures are currently in a subdirectory in the documentation project. If the definitions of the function are in the main documents, I pass "./Bilder/bild.png" as the path and it works.
Now I have packed my functions into a package and installed them locally. Now typst forms an absolute path with the package path, e.g. "/home/nutzer/.local/share/typst/packages/meine/elemente/0.2.1/Bilder/bild.png".
Does anyone have an idea to solve this problem

Thanks, Bernd

**Comment by andreasKroepelin** [comment]
Paths are always relative to the file they're used in. The typical solution is to make the package work with images, not paths, and then pass `#image("path/to/img")` from your document to the package.

**Reply by BerndDemian** [reply]
In my package is the function #let einBild(ueb, bild) = {figure(image(bild), width: 70%), cation:emph(ueb)) ......
When I call from the document  #einBild2("Unterschrift", "Bilder/owens_lila.png"), so typst give the error message

file not found: searched at home/nutzer/.local/share/typst/meine/elemente/0.2.1/Bilder/

I don't want in all document include the same typst-code, where will maintain this? I think that's is the task of a package.
In this moment I have more then ten projects.

Thanks andreas

**Reply by andreasKroepelin** [reply]
In your package, you put

```
#let einBild(ueb, bild) = {
  figure(cation: emph(ueb))[
    #set image(width: 70%)
    #bild
  ]
}
```

and in your document you can then call `#einBild("caption", image("path/to/img"))`

PS: You do you in your code but I'd suggest to keep you questions (including the code) in English so everyone can understand and possibly help you.

PPS: Use backticks to display code in your questions:
````
`inline code` and
```
block code
```
````

becomes

`inline code` and
```
block code
```

**Reply by BerndDemian** [reply]
Thanks, i will tomorrow the code testing!

**Reply by BerndDemian** [reply]
Thanks, for me not optimal, but better than before and when I take caption and not cation, it will be work for me. ;-)

**Comment by laurmaedje** [comment]
It's a bit problematic right now, see https://github.com/typst/typst/issues/1185. There will be a fix at some point, likely involving a path type so that we can make paths relative to where they were specified instead of where they end up being used.

**Reply by BerndDemian** [reply]
in latex there was a command to set the graphic path \graphicspath{path_to_graphics}, this would help or typst recognizes the difference between relatively and absolute paths.

Thanks for the info.
